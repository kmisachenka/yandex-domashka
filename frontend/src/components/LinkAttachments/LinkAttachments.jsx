import React from 'react';
import * as PropTypes from 'prop-types';

import styles from './LinksAttachemnt.module.scss';
import { ReactComponent as UrlSVG } from './svg/url.svg';

function getHostname(url) {
  if (url) {
    return new URL(url).hostname;
  }
  return null;
}

export default function LinkAttachments(props) {
  const { attachments } = props;

  if (attachments.length > 1) {
    return (
      <ul className={styles.list}>
        {attachments.map(({ url }) => (
          <li key={url} className={styles.item}>
            <UrlSVG className={styles.icon} />
            <a
              href={url}
              className={styles.url}
              target="_blank"
              rel="noopener noreferrer"
            >
              {getHostname(url)}
            </a>
          </li>
        ))}
      </ul>
    );
  }
  const { url } = attachments[0];
  return (
    <>
      <UrlSVG className={styles.icon} />
      <a
        href={url}
        className={styles.url}
        target="_blank"
        rel="noopener noreferrer"
      >
        {getHostname(url)}
      </a>
    </>
  );
}

LinkAttachments.propTypes = {
  /* eslint-disable-next-line */
  attachments: PropTypes.array.isRequired,
};
