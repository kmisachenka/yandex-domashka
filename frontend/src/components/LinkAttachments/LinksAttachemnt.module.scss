@import '../../media';

.list {
  @include list-style-none;

  display: inline-flex;
  flex-direction: column;
  justify-content: center;

  width: 100%;
}

.item {
  border-bottom: 1px solid #fff;
  display: inline-flex;
  justify-self: center;
  padding-bottom: 12px;

  padding-top: 12px;

  &:first-child {
    padding-bottom: 12px;
    padding-top: 0;
  }

  &:last-child {
    border-bottom: 0;
    padding-bottom: 0;
  }
}

.icon {
  margin-left: 12px;
  margin-right: 12px;
}

.url {
  color: #000;
  font-family: Helvetica, serif;
  font-size: 16px;
  font-weight: bold;
  line-height: 24px;
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
}
