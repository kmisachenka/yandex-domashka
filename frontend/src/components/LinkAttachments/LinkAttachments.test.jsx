import React from 'react';
import { shallow } from 'enzyme';

import LinkAttachments from './LinkAttachments';

describe('LinkAttachments', () => {
  it('should match snapshot', () => {
    const attachments = [
      {
        type: 'link',
        url: 'https://thiscatdoesnotexist.com',
      },
    ];
    const wrapper = shallow(<LinkAttachments attachments={attachments} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a list when greater then 2', () => {
    const attachments = [
      {
        type: 'link',
        url: 'https://thiscatdoesnotexist.com',
      },
      {
        type: 'link',
        url: 'https://yandex.by',
      },
    ];
    const wrapper = shallow(<LinkAttachments attachments={attachments} />);
    expect(wrapper).toHaveDisplayName('ul');
    const listItems = wrapper.find('li');
    expect(listItems).toHaveLength(2);
  });
  it('should render a link when 1 attachemnt', () => {
    const attachments = [
      {
        type: 'link',
        url: 'https://tut.by',
      },
    ];
    const wrapper = shallow(<LinkAttachments attachments={attachments} />);
    expect(wrapper).not.toHaveDisplayName('ul');
    const links = wrapper.find('a');
    expect(links).toHaveLength(1);
  });
});
