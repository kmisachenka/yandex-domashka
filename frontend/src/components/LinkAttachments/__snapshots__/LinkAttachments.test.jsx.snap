// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkAttachments should match snapshot 1`] = `
<Fragment>
  <ForwardRef
    className="icon"
  />
  <a
    className="url"
    href="https://thiscatdoesnotexist.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    thiscatdoesnotexist.com
  </a>
</Fragment>
`;
